dipy/reconst/dti.py

```diff
@@ -1242,22 +1283,41 @@ def wrapped_fit_tensor(design_matrix, data, step=step,
             size = np.prod(shape)
             step = int(step) or size
             if step >= size:
-                return fit_tensor(design_matrix, data, *args, **kwargs)
+                if return_S0_hat:
```

Comment:
Here as well, consider whether the logic could be simplified by passing along the default value in the case where `return_S0_hat` is set to `False`.